### ISSUE TITLE
Over-running cluster jobs

### ISSUE THREAD

**Issue by andrewjmc** [nextflow-io/nextflow#1728]
## New feature

Allow automatic resubmission of processes which overrun maximum cluster time but can been checkpointed and therefore resumed.

## Usage scenario 

Running nextflow on a maintained cluster, maximum job durations may be 72 hours. Many genome assembly jobs will take longer, but the tools (e.g. metaBAT, metaSPADES) will save checkpoints.

Conventional resume behaviour will not help, as all running tasks will be interrupted, and the resume will recommence the failed jobs afresh.

## Suggest implementation 

Have an option for cluster jobs (e.g. PBSPro) to resubmit within the same working directory if a job did not run to completion.

I am currently trying to do this manually using the feedback loop, but this appears nontrivial for non-singleton inputs, and I am not sure if I can detect a time-overrun job and override the error handling.

**Comment by pditommaso** [comment]
I think it could be done adding an option disabling the creation of a new task work dir when re-trying the execution. Would that work?

**Comment by andrewjmc** [comment]
Yes, I think that would - it would be down to the user (me) to ensure that the script can detect whether there is anything to resume (e.g. metaspades and metabat require different command line options)

**Comment by pditommaso** [comment]
I think the most difficult thing is to find a sensible name for this option, any suggestion?

**Comment by andrewjmc** [comment]
Like naming a child! I can't think of a good name for a boolean variable which fits with other process directives. How about a mode variable?

resubmitMode: "clean" (use a fresh working directory "dirty" (use same working directory)

**Comment by andrewjmc** [comment]
Hi @pditommaso -- any news on this? If you can point me to the right places I might be able to have a go!
